contracts/src/systems/end_game.cairo

```diff
@@ -0,0 +1,49 @@
+#[system]
+mod end_game_system {
+    use dojo::world::Context;
+    use starknet::ContractAddress;
+    use tsubasa::components::{Game, Outcome};
+    use tsubasa::events::{emit, EndGame};
+    use array::ArrayTrait;
+
+    fn execute(ctx: Context, game_id: felt252, scored_player: ContractAddress) {
```

Comment:
Open question, do we call this system after every end turn and if the game shouldn't be ended it just does nothing or should it be called from the `end_turn` system if the conditions are met ?